handler/handler_test.go

```diff
@@ -853,6 +903,264 @@ func TestUnHappyDeleteAnnotationsWhenWritingAnnotationsFails(t *testing.T) {
 	assert.Equal(t, http.StatusInternalServerError, resp.StatusCode)
 }
 
+func TestHappyAddAnnotation(t *testing.T) {
+	rw := new(RWMock)
+	annAPI := new(AnnotationsAPIMock)
+	aug := new(AugmenterMock)
+
+	oldHash := randomdata.RandStringRunes(56)
+	newHash := randomdata.RandStringRunes(56)
+
+	rw.On("Write", mock.AnythingOfType("*context.valueCtx"), "83a201c6-60cd-11e7-91a7-502f7ee26895", &expectedCanonicalisedAnnotationsAfterAdditon, oldHash).Return(newHash, nil)
+	annAPI.On("GetAllButV2", mock.Anything, "83a201c6-60cd-11e7-91a7-502f7ee26895").Return(expectedAnnotations.Annotations, nil)
+
+	h := New(rw, annAPI, annotations.NewCanonicalizer(annotations.NewCanonicalAnnotationSorter), aug, time.Second)
+	r := vestigo.NewRouter()
+
+	r.Post("/drafts/content/:uuid/annotations", h.AddAnnotation)
+
+	ann := annotations.Annotation{"http://www.ft.com/ontology/annotation/mentions", "100e3cc0-aecc-4458-8ebd-6b1fbc7345ed", "", "", "", false}
```

Comment:
We should use the format:
```
	{
		Predicate: "http://www.ft.com/ontology/annotation/mentions",
		ConceptId: "http://www.ft.com/thing/100e3cc0-aecc-4458-8ebd-6b1fbc7345ed",
	}
```
because this way if someone changes the Annotation structure properties' order the test would not break.

Reply:
Fixed